literal/utils.go

```diff
@@ -14,77 +14,77 @@ import (
 	"github.com/substrait-io/substrait-go/v3/types"
 )
 
-func NewBool(value bool) expr.Literal {
-	return expr.NewPrimitiveLiteral[bool](value, false)
+func NewBool(value bool, nullable bool) expr.Literal {
```

Comment:
These changes while straightforward are going to break a lot of existing code.  Two alternates would be to add an additional function with the new behavior (NewNullBool creating a nullable Bool or NewNullableBool with the option similar to what is implemented here).  A third is to actually make this a breaking change which would cause a version bump.  A fourth option would be to add a new method to expr.Literal that would change the nullability to be set.

Reply:
will mark this as breaking change